### DISCUSSION TITLE
Proposal: activation via single global hotkey

### DISCUSSION THREAD

**Discussion by mikaraunio** [rxhanson/Rectangle#381] [Ideas]
I'd love to have the possibility to activate Rectangle via a single hotkey. Window manipulation would then be carried out via simple keystrokes (for example, modifier-less arrow keys to move windows), until a key such as Escape is pressed to exit "Rectangle-mode".

The current command structure with dedicated shortcut keys for each function:
* makes it hard to remember all shortcuts
* requires extra finger dexterity with shortcuts using multiple modifiers
* is prone to conflicts especially if Rectangle's config is shared between multiple systems with different apps installed

With only a single global hotkey, the possibility of conflicts would be greatly reduced. Commands would require fewer modifiers and thus be easier to type and to remember.

**Comment by rxhanson** [comment]
Thanks for the idea!

My preferred implementation of this concept is what I have implemented in Hookshot (https://hookshot.app). The concept there is holding one modifier key combination and moving the mouse cursor in a direction. Of course, this requires taking hands off the keyboard, but I find it to still be faster and more convenient for me. 

Toggling a "mode" would also require Rectangle (or Hookshot) to listen to all keystrokes. Right now it just registers your shortcuts with macOS and only keys off of them, and I like this better from a developer and user perspective. Just an opinion, but I have always found keyboard mode toggles annoying (accidentally having insert or caps lock active unknowingly).

For keyboard-only improvements, Hookshot also allows you to configure the same state machine behavior as windows, where you can get a top left corner with mod + right then mod + up, etc. Additionally, it has the make-your-own shortcut tab where you can create custom shortcuts and their repeated actions. That way you can get whatever flow you want just repeating the same keyboard shortcut(s). Here's a post I wrote about the custom shortcut builder: https://medium.com/ryan-hanson/custom-window-dimension-shortcuts-in-macos-e7df347dfd24

**Reply by mikaraunio** [reply]
Thanks very much for your feedback and consideration. Very much agree that UI modality is a matter of (often strong) opinion, looks like we're at pretty opposite ends of the spectrum on this one :) Guess I feel the same way about accidentally pressing Ctrl-Shift instead of Cmd-Ctrl as you do about inadvertently having Caps Lock on.

With a judiciously chosen shortcut, I don't think accidental activation would be much of an issue, but totally understand if that's not a direction you'd want to take Rectangle to.

The Hookshot custom shortcut builder looks very cool even if it's not quite what I'm after here.
